File: Node/app/src/models/review.js
import { Sequelize } from "sequelize"; 
import db from "../../config/config.js";
 
const { DataTypes } = Sequelize;

const Review = db.define('reviews', {
  idRating: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  userId: {
    type: DataTypes.INTEGER
  },
  movieId: {
    type: DataTypes.INTEGER
  },
  ratingStars: {
    type: DataTypes.INTEGER
  },
  textComment: {
    type: DataTypes.STRING
  },
  movieTitle: {
    type: DataTypes.STRING
  },
  movieImg: {
    type: DataTypes.STRING
  },
  movieBackground: {
    type: DataTypes.STRING
  },
  movieDescription: {
    type: DataTypes.STRING
  },
  movieLanguage: {
    type: DataTypes.STRING
  },
  movieReleaseDate: {
    type: DataTypes.STRING
  },
  movieVoteAverage: {
    type: DataTypes.INTEGER
  },
  timestamp: {
    type: DataTypes.DATE, 
    defaultValue: Sequelize.literal('CURRENT_TIMESTAMP')
  }

}, {
  freezeTableName: true,
  timestamps: true
});
 
export default Review;

File: SpringBoot/src/main/java/com/thenetvalue/usersManagement/model/Review.java
package com.thenetvalue.usersManagement.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import java.time.LocalDateTime;
import java.util.Date;

@Setter
@Getter
@Entity
@Table(name = "reviews")
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator = "reviewSequence")
    @SequenceGenerator(name = "reviewSequence", sequenceName = "reviews_seq", allocationSize = 1)
    private int id;

    private int userId;

    private int movieId;

    private int ratingStars;

    private String textComment;

    private String movieTitle;

    private String movieImg;

    private String movieBackground;

    @Column(length = 1000)
    private String movieDescription;

    private String movieLanguage;

    private String movieReleaseDate;

    private float movieVoteAverage;
    @CreationTimestamp
    @Column(name = "timestamp", nullable = false, updatable = false)
    private LocalDateTime timestamp = LocalDateTime.now();
}
